src/tactic/linarith/parsing.lean

```diff
@@ -97,13 +109,18 @@ meta def linear_form_of_expr (red : transparency) : exmap → expr → tactic (e
       (m', comp2) ← linear_form_of_expr m' e2,
       return (m', comp1.add (comp2.scale (-1)))
 | m `(-%%e) := do (m', comp) ← linear_form_of_expr m e, return (m', comp.scale (-1))
+| m p@`(%%e ^ %%n) :=
```

Comment:
Note that `expr.to_nat` doesn't check at all that the expression is a natural number, it could also be a real, etc.

```suggestion
| m p@`(%%e ^ (%%n : ℕ)) :=
```
No idea whether this works though.

Reply:
That doesn't work, but ``p@`(@has_pow.pow _ ℕ _ %%e %%n)`` does.